packages/common/src/hooks/purchaseContent/usePurchaseContentErrorMessage.ts

```diff
@@ -0,0 +1,37 @@
+import { BuyUSDCErrorCode } from 'store/index'
+import {
+  PurchaseContentErrorCode,
+  PurchaseErrorCode
+} from 'store/purchase-content'
+import { formatPrice } from 'utils/formatUtil'
+
+import { RemoteVarHook } from '../useRemoteVar'
+
+import { useUSDCPurchaseConfig } from './useUSDCPurchaseConfig'
+
+const messages = {
+  generic: 'Your purchase was unsuccessful.',
+  minimumPurchase: (minAmount: number) =>
+    `Total purchase amount must be at least $${formatPrice(minAmount)}.`,
+  maximumPurchase: (maxAmount: number) =>
+    `Total purchase amount may not exceed $${formatPrice(maxAmount)}.`
+}
+
+export const usePurchaseContentErrorMessage = (
+  errorCode: PurchaseContentErrorCode,
+  useRemoteVar: RemoteVarHook
```

Comment:
I really want a better pattern for this. Having to pass the remote var hook in from arbitrary places is at best an obnoxious dependency injection pattern 😁

Reply:
yeah gross. probs fine though, it's at least understandable.
but also, why not just have a whole USDCPurchaseConfig passed here as an arg and the caller does

```
const USDCPurchaseConfig = useUSDCPurchaseConfig()
const errorMsg = usePurchaseContentErrorMessage(errorCode, USDCPurchaseConfig)
```

Reply:
Good suggestion. I followed that pattern in a previously PR and ended up getting rid of it in favor of this arrangement. It ends up being about the same amount of effort (just changes where you call the hook, but you're still passing something in).
This is more easily adaptable to moving the import of the hook into common, which is the eventual plan.